examples/pytorch/seal/sampler.py

```diff
@@ -0,0 +1,294 @@
+import torch
+import dgl
+from torch.utils.data import DataLoader, Dataset
+from dgl import DGLGraph, NID, EID
+from utils import drnl_node_labeling, coalesce_graph
+from dgl.dataloading.negative_sampler import Uniform
+from dgl import add_self_loop
+import os.path as osp
+from tqdm import tqdm
+from copy import deepcopy
+
+
+class GraphDataSet(Dataset):
+    """
+    GraphDataset for torch DataLoader
+    """
+
+    def __init__(self, graph_list, tensor):
+        self.graph_list = graph_list
+        self.tensor = tensor
+
+    def __len__(self):
+        return len(self.graph_list)
+
+    def __getitem__(self, index):
+        return (self.graph_list[index], self.tensor[index])
+
+
+class SEALDataLoader(object):
+    """
+    Data Loader of SEAL
+    Attributes:
+        dataset(Dataset): dataset
+        batch_size(int): size of batch
+    """
+
+    def __init__(self, dataset, batch_size, num_workers=1, shuffle=True,
+                 drop_last=False, pin_memory=False):
+        self.total_graphs = len(dataset)
+        self.dataloader = DataLoader(dataset=dataset, collate_fn=self._collate, batch_size=batch_size, shuffle=shuffle,
+                                     num_workers=num_workers, drop_last=drop_last, pin_memory=pin_memory)
+
+    def _collate(self, batch):
+        batch_graphs, batch_labels = map(list, zip(*batch))
+
+        batch_graphs = dgl.batch(batch_graphs)
+        batch_labels = torch.stack(batch_labels)
+        return batch_graphs, batch_labels
+
+    def __len__(self):
+        """Return the number of batches of the data loader."""
+        return len(self.dataloader)
+
+    def __iter__(self):
+        return iter(self.dataloader)
+
+
+class PosNegEdgesGenerator(object):
```

Comment:
Should this Generator be accelerated with `DataLoader`?

Reply:
I think there is no need for accelerating the Generator, since generating all the 2500000 negative edges cost less than 0.1s.